object_store/src/aws/mod.rs

```diff
@@ -472,6 +623,50 @@ impl AmazonS3Builder {
         self
     }
 
+    /// Set an option on the builder via a key - value pair.
+    pub fn with_option(
+        mut self,
+        key: impl Into<String>,
```

Comment:
```suggestion
        key: impl AsRef<str>
```
We don't actually need an owned string